modules/core/src/main/java/org/apache/ignite/cluster/ClusterState.java

```diff
@@ -29,4 +30,43 @@
 
     /** Cluster activated. Cache read operation allowed, Cache data change operation aren't allowed. */
     ACTIVE_READ_ONLY;
+
+    /** Enumerated values. */
+    private static final ClusterState[] VALS = values();
+
+    /**
+     * Efficiently gets enumerated value from its ordinal.
+     *
+     * @param ord Ordinal value.
+     * @return Enumerated value or {@code null} if ordinal out of range.
+     */
+    @Nullable public static ClusterState fromOrdinal(int ord) {
```

Comment:
why returning null preferable than Optional ?

Reply:
It's a common practice in the Ignite project. Look at other enums